ckanext/switzerland/dcat/profiles.py

```diff
@@ -34,97 +40,120 @@ def _object_value(self, subject, predicate, multilang=False):
+            publisher = {'label': self._object_value(agent, RDFS.label)}
+            publishers.append(publisher)
 
-        publisher = {}
+        return publishers
 
-        for agent in self.g.objects(subject, predicate):
+    def _relations(self, subject, predicate):
 
-            publisher['uri'] = (str(agent) if isinstance(agent,
-                                rdflib.term.URIRef) else None)
+        relations = []
 
-            publisher['name'] = self._object_value(agent, RDFS.label)
-            publisher['email'] = None
-            publisher['url'] = None
-            publisher['type'] = None
+        for relation_node in self.g.objects(subject, predicate):
+            relation = {
+                'label': self._object_value(relation_node, RDFS.label),
+                'url': relation_node
+            }
+            relations.append(relation)
 
-        return publisher
+        return relations
 
-    def _contact_details(self, subject, predicate):
-        '''
-        Returns a dict with details about a vcard expression
+    def _keywords(self, subject, predicate):
 
-        Both subject and predicate must be rdflib URIRef or BNode objects
+        keywords = {}
 
-        Returns keys for uri, name and email with the values set to
-        None if they could not be found
-        '''
+        for lang in get_langs():
+            keywords[lang] = []
 
-        contact = {}
+        for keyword_node in self.g.objects(subject, predicate):
+            keywords[keyword_node.language].append(unicode(keyword_node))
 
-        for agent in self.g.objects(subject, predicate):
+        return keywords
+
+    def _contact_points(self, subject, predicate):
+
+        contact_points = []
+
+        for contact_node in self.g.objects(subject, predicate):
+            email = self._object_value(contact_node, VCARD.hasEmail)
+            email_clean = email.replace('mailto:', '')
+            contact = {
+                'name': self._object_value(contact_node, VCARD.fn),
+                'email': email_clean
+            }
+
+            contact_points.append(contact)
 
-            contact['uri'] = (str(agent) if isinstance(agent,
-                              rdflib.term.URIRef) else None)
+        return contact_points
 
-            contact['name'] = self._object_value(agent, VCARD.fn)
+    def _temporals(self, subject, predicate):
 
-            contact['email'] = self._object_value(agent, VCARD.hasEmail)
+        temporals = []
 
-        return contact
+        for temporal_node in self.g.objects(subject, predicate):
+            start_date = self._object_value(temporal_node, SCHEMA.startDate)
+            end_date = self._object_value(temporal_node, SCHEMA.endDate)
+            if start_date or end_date:
+                temporals.append({
+                    'start_date': self._clean_datetime(start_date),
+                    'end_date': self._clean_datetime(end_date)
```

Comment:
You should use a try/except here, as `_clean_datetime` raises a ValueError. It's possible to only have a start or an end date. Alternatively you could set a default value like `None` in `_clean_datetime` instead of raising an exception.